solidity/contracts/peripherals/Auction.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: UNICENSE
+pragma solidity 0.8.30;
+uint256 constant AUCTION_TIME = 1 weeks;
+
+contract Auction {
+	mapping(address => uint256) public purchasedRep;
+	uint256 public totalRepPurchased;
+	uint256 public repAvailable;
+	uint256 public auctionStarted;
+	uint256 public ethAmountToBuy;
+	bool public finalized;
+	address immutable owner;
+
+	event Participated(address user, uint256 repAmount, uint256 ethAmount, uint256 totalRepPurchased);
+	event FinalizedAuction(address user, uint256 repAmount, uint256 ethAmount);
+	event AuctionStarted(uint256 ethAmountToBuy, uint256 repAvailable);
+
+	constructor(address _owner) {
+		owner = _owner;
+	}
+
+	function participate(uint256 repToBuy) public payable {
+		require(auctionStarted > 0, 'Auction needs to have started');
+		require(!finalized, 'Already finalized');
+		require(msg.value > 0, 'need to invest with eth!');
```

Comment:
Using ETH instead of WETH means we can't use the name contract for other tokens.